### ISSUE TITLE
BN error

### ISSUE THREAD

**Issue by fzh-adham** [MaximIntegratedAI/ai8x-synthesis#350]
hi 
when I want to generate  c files I get this error "the checkpoint file contains 1-dimensional weights for `conv1.bn.weight` with dimensions (24,). Ensure the BatchNorm layers have been folded."
what is the problem and what is the solution?

**Comment by ermanok** [comment]
Hello,

The hardware does not support BatchNorm (BN) layer so if you want to use a BN layer in your model, you have to fuse these layers to the preceding convolution layers (see [here](https://github.com/analogdevicesinc/ai8x-training?tab=readme-ov-file#batch-normalization)). If you enable Quantization Aware Training (QAT) during your model training, the code automatically folds the the batchnorm layers. So if you use `qat_checkpoint.pth.tar` or `qat_best.path.tar`, you will not get such an error during the code synthesis. If you want to use post training quantization without QAT, you may run `batchnormfuser.py` to fold the BN layers to the convolutional layers.

**Comment by fzh-adham** [comment]
Hi
Thanks for reply
Yes,I have trained in both ways and I just have problem in none qat
So for your second choice in command how do I run my model in training
with  batchnormfuser.py
Could you please tell me in details🙏🙏

**Comment by ermanok** [comment]
It is a python script that you can run from the command line. The details of the batchnormfuser is [here](https://github.com/analogdevicesinc/ai8x-training?tab=readme-ov-file#command-line-arguments-1). But we always suggest to use QAT as the direct quantization of a floating model may cause huge accuracy drop.

**Comment by fzh-adham** [comment]
Thanks a lot

I have another problem in command

In the documentation is suggested to use --energy for energy during
processing but in my max78000fthr board it doesn't show

How can I get energy ?

**Comment by ermanok** [comment]
You need a [MAX78000 EvKit](https://www.analog.com/en/resources/evaluation-hardware-and-software/evaluation-boards-kits/max78000evkit.html) to measure and observe the energy consumption. The feather board does not have an energy measurement circuitry.

**Comment by fzh-adham** [comment]
thanks for reply 
is there any way to calculate energy consumption and power?

**Comment by ermanok** [comment]
There is no other straightforward approach to measure the energy consumption.
